### PR TITLE
8253457: Remove unimplemented register stack functions

### PR DESCRIPTION
**Pull request by coleenp**
Please review removed functions left over from Itanium.

Ran tier1 testing on Oracle platforms (linux-x64, macos-x64, windows-x64 and linux-aarch64) and built on linux-arm32,linux-ppc64le-debug,linux-s390x-debug,linux-x64-zero.

Thanks,
Coleen

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253457](https://bugs.openjdk.java.net/browse/JDK-8253457): Remove unimplemented register stack functions

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/300/head:pull/300`
`$ git checkout pull/300`

### CODE REVIEW

File: src/hotspot/os_cpu/linux_arm/thread_linux_arm.hpp [hunk 1]
```diff
@@ -40,26 +40,10 @@
   frame pd_last_frame();
 
  public:
-  intptr_t* last_Java_fp()                       { return _anchor.last_Java_fp(); }
-  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);  }
-  void  set_last_Java_pc(address pc)             { _anchor.set_last_Java_pc(pc);  }
-
```
Comment (iklam) [inline src/hotspot/os_cpu/linux_arm/thread_linux_arm.hpp:46]: Why is this remove from _arm, but not for thread_linux_aarch64.hpp

**Review by iklam — CHANGES_REQUESTED** [review]
Looks good overall to me. I can see all these functions with no callers. Should they also be removed? I saw you removed only for thread_linux_arm.hpp.

```
./os_cpu/linux_aarch64/thread_linux_aarch64.hpp:  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);   }
./os_cpu/linux_x86/thread_linux_x86.hpp:  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);   }
./os_cpu/windows_x86/thread_windows_x86.hpp:  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);   }
./os_cpu/linux_arm/thread_linux_arm.hpp:  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);  }
./os_cpu/bsd_x86/thread_bsd_x86.hpp:  void  set_last_Java_fp(intptr_t* fp)           { _anchor.set_last_Java_fp(fp);   }
./cpu/aarch64/javaFrameAnchor_aarch64.hpp:  void set_last_Java_fp(intptr_t* fp)                { OrderAccess::release(); _last_Java_fp = fp; }
./cpu/arm/javaFrameAnchor_arm.hpp:  void set_last_Java_fp(intptr_t* fp)                { _last_Java_fp = fp; }
./cpu/x86/javaFrameAnchor_x86.hpp:  void set_last_Java_fp(intptr_t* fp)                { _last_Java_fp = fp; }

./share/runtime/javaFrameAnchor.hpp:  void set_last_Java_pc(address pc)              { _last_Java_pc = pc; }
./os_cpu/linux_arm/thread_linux_arm.hpp:  void  set_last_Java_pc(address pc)             { _anchor.set_last_Java_pc(pc);  }
```

I am not sure about `last_Java_fp()` as I didn't check thoroughly.

The other 5 removed function seem OK to me -- they don't do anything. The only exception is the *_zero.hpp versions that had some asserts, but I guess these asserts never caught anything since they don't exist on other ports anyway.

File: src/hotspot/os_cpu/linux_arm/thread_linux_arm.hpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/os_cpu/linux_arm/thread_linux_arm.hpp:46]: Oops, I didn't mean to remove these ones.

I was focusing on the stack related functions. but I can file an RFE to remove these next.  Thanks!

**Review by iklam — APPROVED** [review]
LGTM

**Review by dholmes-ora — APPROVED** [review]
Good cleanup! Thanks.

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**